src/Git/CommitJob.php

```diff
@@ -26,4 +27,12 @@ public function handle()
     {
         Git::as($this->committer)->commit($this->message);
     }
+
+    /**
+     * @return array<int, object>
+     */
+    public function middleware(): array
+    {
+        return [(new WithoutOverlapping('git'))->expireAfter(60)->releaseAfter(30)];
```

Comment:
@jesseleite not sure what to do for the 'key' as we want it common so there is no overlap.

Reply:
Maybe `statamic-git` as key, only because there could technically be other job middleware in custom-Laravel-land? Likely would never be a problem either way though.